compiler/native/environment.go

```diff
@@ -109,6 +109,58 @@ func (c *client) EnvironmentServices(s yaml.ServiceSlice) (yaml.ServiceSlice, er
 	return s, nil
 }
 
+// EnvironmentSecrets injects environment variables
+// for each secret plugin in a yaml configuration.
+func (c *client) EnvironmentSecrets(s yaml.SecretSlice) (yaml.SecretSlice, error) {
```

Comment:
This function is a combination of existing Steps and Services environment functions to ensure all necessary environment configuration is injected